e310x-hal/src/gpio.rs

```diff
@@ -4,13 +4,46 @@ use core::marker::PhantomData;
 
 use portable_atomic::{AtomicU32, Ordering};
 
+/// Event Type for GPIO interrupts.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum EventType {
+    /// High level event
+    High,
+    /// Low level event
+    Low,
```

Comment:
Maybe a `BothLevels` variant would be useful for disabling interrupts (and completion)

```suggestion
    Low,
    /// Both levels event
    BothLevels,
```